src/app/content/highlights/hooks/closeMyHighlights.ts

```diff
@@ -0,0 +1,21 @@
+import { push } from '../../../navigation/actions';
+import { query } from '../../../navigation/selectors';
+import { removeParamFromQuery } from '../../../navigation/utils';
+import { ActionHookBody } from '../../../types';
+import { actionHook } from '../../../utils';
+import { modalQueryParameterName } from '../../constants';
+import { getContentParams } from '../../utils/urlUtils';
+import { closeMyHighlights } from '../actions';
+
+export const hookBody: ActionHookBody<typeof closeMyHighlights> = ({
```

Comment:
seems like we could have a generic `closeModal` hook body that we can just attach to different actions, i'm pretty sure you can get all the params you need out of the navigation state without any context specific selectors. 

also, if the opening if the modal always triggers a `push`, then we should be able to safely close it with a `pop` without worrying about all this other stuff right?